include/aos/common/tools/map.hpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_MAP_HPP_
+#define AOS_MAP_HPP_
+
+#include "aos/common/tools/array.hpp"
+#include "aos/common/tools/utils.hpp"
+
+namespace aos {
+
+/**
+ * Simple map implementation based on a non sorted array.
+ */
+template <typename Key, typename Value>
+class Map {
+public:
+    /**
+     * Returns a reference to the value with specified key. And an error if a key is absent.
+     *
+     * @param key
+     * @return RetWithError<Value&>
+     */
+    RetWithError<Value&> At(const Key& key);
+
+    /**
+     * Returns reference to the value with specified key. And an error if a key is absent.
+     *
+     * @param key
+     * @return RetWithError<const Value&>
+     */
+    RetWithError<const Value&> At(const Key& key) const;
+
+    /**
+     * Replaces map with elements from the list.
+     *
+     * @param list source list
+     * @return Error
+     */
+    Error Assign(const std::initializer_list<Pair<Key, Value>>& list);
```

Comment:
Unfortunatelly no std in `aos_core_lib_cpp`. We can replace this method with one:

` Error Assign(const Array<Pair<Key, Value>> list);` to assign from array.

Reply:
fixed